java/apirag-core/src/main/java/com/datasqrl/ai/models/ChatProviderFactory.java

```diff
@@ -0,0 +1,12 @@
+package com.datasqrl.ai.models;
+
+import com.datasqrl.ai.backend.FunctionBackend;
+import org.apache.commons.configuration2.Configuration;
+
+public interface ChatProviderFactory {
```

Comment:
Do we need this interface if you already have a base `AbstractChatProviderFactory`?

Reply:
Yes, this is needed for the dynamic service loading of the factory classes.

In general, I like the pattern of having an interface that defines the contract and an abstract class if I want to consolidate some generic implementation details that are the same for all interface implementations. Makes it easier to understand what the contract is and what is implementation specific.